### DISCUSSION TITLE
Can you run just the server / headless?

### DISCUSSION THREAD

**Discussion by leggt** [jdno/auto-traffic-control#171] [Q&A] [answered]
This is really cool! One question though, if I wanted to run this headless, just the server without a gui, on my remote server, just for playing around with the messages, could I do that? I actually tried to spin this up in a docker container (https://github.com/jlesage/docker-baseimage-gui) but ran into an error where bevy is looking for a hardware GPU.

**Comment by jdno** [comment, marked as answer]
Hi @leggt!

Sadly, running a headless server is currently not supported. But the Bevy engine theoretically [supports](https://github.com/bevyengine/bevy/blob/main/examples/app/headless.rs) it, and it has been on the long list of features that I eventually want to implement.

My guess is that headless mode will require a bit of refactoring, since the implementation currently mixes simulation and rendering. Just disabling the default plugins will probably not be enough.

Can you share your Dockerfile or the error message? Maybe there is a quick fix for it...

**Reply by leggt** [reply]
Hey @jdno thanks for the response. I eventually spun up a VM to run auto-traffic-control in and that supports my use case. 

For curiosity, I never got docker to work; I think it's some combination of me not having all the libraries installed or not mounting all the hardware I need into the container. It probably isn't worth pursuing much but here's the Dockerfile I was using..

```
# https://github.com/jlesage/docker-baseimage-gui
FROM jlesage/baseimage-gui:debian-11

RUN apt-get update -y; apt-get install -y gcc make build-essential curl git libasound2-dev pkg-config libudev-dev protobuf-compiler


# baseimage-gui is going to run an app as the app user, so install all the rust stuff under that user
WORKDIR /
RUN useradd -u 1000 -ms /bin/bash app; mkdir /atc; chown app /atc
USER app
RUN curl https://sh.rustup.rs -sSf | sh -s -- -y
RUN git clone https://github.com/jdno/auto-traffic-control.git atc
WORKDIR /atc
ENV PATH="/home/app/.cargo/bin:$PATH"
RUN cargo fetch; cargo build

USER root
# More stuff I found I needed
RUN apt-get update; apt-get install -y x11-xserver-utils mesa-vulkan-drivers

COPY startapp.sh /startapp.sh
```

startapp.sh is
```
#!/bin/bash
cd /atc
export HOME=/home/app
export PATH="$HOME/.cargo/bin:$PATH"
cargo run game
```
